### PR TITLE
New Microsoft.Logic API Version: 2018-07-01-preview

### PR DESCRIPTION
**Pull request by refortie**
## Description

REST spec PRs:
[Add new microsoft.logic version](https://github.com/Azure/azure-rest-api-specs/pull/3683)
[Change some string Microsoft.Logic modeled enums back into enums](https://github.com/Azure/azure-rest-api-specs/pull/3775)
[Remove default response from Sessions](https://github.com/Azure/azure-rest-api-specs/pull/3868)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by refortie** [comment]
This introduces a lot of breaking changes. We are doing a version up of the APIs

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj [hunk 1]
```diff
@@ -27,4 +27,16 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+
+  <ItemGroup>
+    <None Update="TestData\IntegrationAccountAssemblyContent.dll">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+    <None Update="TestData\IntegrationAccountCertificate.cer">
```
Comment (dsgouda) [inline src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: Is this an authentication certificate? Not sure if we can check this in 
@shahabhijeet  FYI

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj [hunk 2]
```diff
@@ -27,4 +27,16 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+
```
Comment (dsgouda) [inline src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: Please remove `TargetFrameworks` from this file
Add ` <TreatWarningsAsErrors>true</TreatWarningsAsErrors>` and fix errors if any are reported

File: src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 	<PropertyGroup>
 		<Description>Microsoft Azure LogicApps Management Library</Description>
-		<Version>3.1.0</Version>
+		<Version>4.0.0</Version>
 		<PackageId>Microsoft.Azure.Management.Logic</PackageId>
 		<PackageTags>Microsoft Azure LogicApps management;LogicApps;LogicApps management;</PackageTags>
 		<PackageReleaseNotes/>
```
Comment (dsgouda) [inline src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj]: Please add `PackageReleaseNotes` here

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
Please fix the CI errors reported

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj (hunk 2 above)
Comment (refortie) [reply src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: Changed

File: src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj (hunk 3 above)
Comment (refortie) [reply src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj]: Added, please let me know if the formatting is incorrect

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj (hunk 1 above)
Comment (refortie) [reply src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: This was a file that had already existed in the repo to be used in tests. I'm not sure how the tests worked without the file being included. They don't work locally when I don't include it. In general this isn't a cert to anything in particular, just a cert in general.

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj [hunk 4]
```diff
@@ -5,10 +5,12 @@
     <Description>Logic.Tests Class Library</Description>
     <DelaySign>true</DelaySign>
     <AssemblyName>Logic.Tests</AssemblyName>
+    <PackageTags />
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|netcoreapp2.0|AnyCPU'">
```
Comment (dsgouda) [inline src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: Please remove these conditions

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Review by dsgouda — CHANGES_REQUESTED** [review]
Need an approval from @shahabhijeet regarding the cert

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj (hunk 1 above)
Comment (shahabhijeet) [reply src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: @refortie if this is a self-signed cert, this should be fin.

File: src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj (hunk 4 above)
Comment (refortie) [reply src/SDKs/Logic/Logic.Tests/Logic.Tests.csproj]: Changed

File: src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj [hunk 5]
```diff
@@ -5,10 +5,52 @@
...
 	<PropertyGroup>
 		<Description>Microsoft Azure LogicApps Management Library</Description>
-		<Version>3.1.0</Version>
+		<Version>4.0.0</Version>
 		<PackageId>Microsoft.Azure.Management.Logic</PackageId>
 		<PackageTags>Microsoft Azure LogicApps management;LogicApps;LogicApps management;</PackageTags>
-		<PackageReleaseNotes/>
+		<PackageReleaseNotes>
+		<![CDATA[
+	  - Updated API Version to 2018-07-01-preview
```
[the remark this answers is not in the document]
Comment (refortie) [reply src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj]: I've cut it down a bit. It won't be an exhaustive list of all the breaking changes anymore, if that is alright?

[the remark this answers is not in the document]
Comment (dsgouda) [reply src/SDKs/Logic/Management.Logic/Microsoft.Azure.Management.Logic.csproj]: This is perfect

**Review by dsgouda — APPROVED** [review]
LGTM, will let @shahabhijeet take a look and we are good to merge

**Comment by dsgouda** [comment]
Talked to @shahabhijeet , self signed certificates are acceptable, thanks!

**Pull request merged**